core/templates/dev/head/pages/state_editor/state_content_editor_directive.html

```diff
@@ -37,4 +50,55 @@
     <button type="button" class="btn btn-default pull-right" ng-click="cancelEdit()">Cancel</button>
     <div style="clear: both;"></div>
   </div>
+  <div ng-if="isCardHeightLimitReached() && cardHeightLimitWarningIsShown">
+    <span class="oppia-card-height-limit-warning">
+      <i class="material-icons oppia-hide-card-height-warning-icon" uib-tooltip="Hide warning" ng-click="hideCardHeightLimitWarning()">&#xE88F;</i>
+      <span translate="I18N_CARD_HEIGHT_LIMIT_WARNING_MESSAGE"></span>
+    </span>
+  </div>
 </div>
+
+<style>
+  .oppia-card-height-limit-warning {
+    color: #0277BD;
+    padding-top: 2px;
+  }
+
+  .shadow-preview-card {
+    left: -30000px;
+    max-width: 560px;
+    position: fixed;
+    top: -30000px;
+  }
+
+
+  /* These rules must be kept in sync with corresponding rules in oppia.css
```

Comment:
Why can't we just use the existing CSS rules in the preview tab?

If that CSS is used more generally then we should move the CSS in that tab out, perhaps to oppia.css. Should not duplicate it here (and in general please try to avoid duplication in the initial PR, don't ever leave it as a TODO).

Reply:
There are a few other places where the we have comments to have css in sync. The links are in this issue: https://github.com/oppia/oppia/issues/6089.

So, would it be better to create a separate PR to move the common rules to `oppia.css`?

Reply:
@DubeySandeep what do you suggest?

Reply:
Hi @DubeySandeep @seanlip, I have issued PR #6107 to fix #6089. We can let that go in before this one and then make the corresponding changes here too. What do you think?

Reply:
Hi @ankita240796 -- that's great, thanks! What you've done is the best way to do it -- I was meaning to write that earlier, but didn't get a chance to do so, so I'm very glad that you went ahead and did it that way :) It's definitely the best and cleanest way to structure it.

I'm about to go to bed, though, so I'll defer the review to @DubeySandeep. But thanks again for doing that!

Reply:
You're welcome @seanlip :smile:

Reply:
@ankita240796, FYI, #6107 is merged so you can continue working on this issue! :)

Reply:
Done, PTAL!